packages/ssz/src/core/createValue.ts

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import assert from "assert";
+import {AnySSZType, FullSSZType, parseType, Type} from "@chainsafe/ssz-type-schema";
+
+import {assertValidValue} from "./assertValidValue";
+import {defaultValue} from "./defaultValue";
+
+/**
+ * Given an ssz type, create an empty value, optionally filled in with a default value
+ *
+ * The default value can be only partially specified.
+ */
+export function createValue<T>(type: AnySSZType<T>, value: any = null): T {
+  return _createValue(parseType(type), value) as T;
```

Comment:
Correct me if im wrong, but it seems like we only fill in with default value when `value === null`, so we can remove a lot of code in `_createValue` if we check if null here and just return `defaultValue` here and remove the checks in `_createValue`

Reply:
:+1:

Reply:
Unfortunately, we still need the null checks in the inner function. The intended behavior is that a user can enter a partially filled object, and the unspecified parts are filled in with default values. Since the inner function is called recursively, those calls to unspecified parts will still need the null check.